src/Tizen.NUI/src/public/Events/GestureDetectorManager.cs

```diff
@@ -0,0 +1,156 @@
+    private TapGestureDetector mTapGestureDetector;
+    private LongPressGestureDetector mLongGestureDetector;
+    private PinchGestureDetector mPinchGestureDetector;
+    private PanGestureDetector mPanGestureDetector;
+
+    /// <summary>
+    ///  Creates a GestureDetectorManager with the user listener.
+    /// </summary>
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public GestureDetectorManager(View view, GestureListener listener)
+    {
+      if(view == null)
+      {
+        throw new global::System.ArgumentNullException(nameof(view));
+      }
+      if(listener == null)
+      {
+        throw new global::System.ArgumentNullException(nameof(listener));
+      }
+
+      mListener = listener;
+      view.GrabTouchAfterLeave = true;
+      init(view);
+    }
+
+    private void init(View view)
+    {
+      mTapGestureDetector = new TapGestureDetector();
+      mLongGestureDetector = new LongPressGestureDetector();
+      mPanGestureDetector = new PanGestureDetector();
+      mPinchGestureDetector = new PinchGestureDetector();
+
+      mTapGestureDetector.Attach(view);
+      mLongGestureDetector.Attach(view);
+      mPanGestureDetector.Attach(view);
+      mPinchGestureDetector.Attach(view);
+    }
+
+    private void InternalOnTap(object sender, TapGestureDetector.DetectedEventArgs e)
+    {
+      mListener.OnTap(sender, e);
+      mTapGestureDetector.Detected -= InternalOnTap;
+    }
+
+    private void InternalOnLongPress(object sender, LongPressGestureDetector.DetectedEventArgs e)
+    {
+      mListener.OnLongPress(sender, e);
+      mLongGestureDetector.Detected -= InternalOnLongPress;
+    }
+
+    private void InternalOnPan(object sender, PanGestureDetector.DetectedEventArgs e)
+    {
+      mListener.OnPan(sender, e);
+      mPanGestureDetector.Detected -= InternalOnPan;
+    }
+
+    private void InternalOnPinch(object sender, PinchGestureDetector.DetectedEventArgs e)
+    {
+      mListener.OnPinch(sender, e);
+      mPinchGestureDetector.Detected -= InternalOnPinch;
+    }
+
+    /// <summary>
+    /// Gestures also work only when there is a touch event.
+    /// </summary>
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public void FeedTouchEvent(object sender, View.TouchEventArgs e)
+    {
+      mTapGestureDetector.Detected -= InternalOnTap;
+      mLongGestureDetector.Detected -= InternalOnLongPress;
+      mPanGestureDetector.Detected -= InternalOnPan;
+      mPinchGestureDetector.Detected -= InternalOnPinch;
+
+      if(e != null &&
+          (e.Touch.GetState(0) != PointStateType.Finished ||
+           e.Touch.GetState(0) != PointStateType.Up       ||
+           e.Touch.GetState(0) != PointStateType.Interrupted))
+      {
+        mTapGestureDetector.Detected += InternalOnTap;
+        mLongGestureDetector.Detected += InternalOnLongPress;
+        mPanGestureDetector.Detected += InternalOnPan;
+        mPinchGestureDetector.Detected += InternalOnPinch;
+      }
+    }
+  }
+
+}
```

Comment:
From what I heard, there need to add a simple empty line in C# :
https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline
Could you add one empty line in both files?